.circleci/cimodel/pytorch_build_definitions.py

```diff
@@ -0,0 +1,369 @@
+    HiddenConf("pytorch_doc_push", parent_build=xenial_parent_config),
+]
+
+
+xenial_parent_config.dependent_tests = xenial_dependent_configs
+
+
+# TODO this hierarchy is a work in progress
+CONFIG_TREE_DATA = [
+    ("trusty", [
+        ("py2.7.9", []),
+        ("py2.7", []),
+        ("py3.5", []),
+        ("py3.6", [
+            ("gcc4.8", []),
+            ("gcc5.4", [False, True]),
+            ("gcc7", []),
+        ]),
+        ("pynightly", []),
+    ]),
+    ("xenial", [
+        ("clang", [
+            ("X", [("py3", [])]),
+        ]),
+        ("cuda", [
+            ("8", [("3.6", [])]),
+            ("9", [
+                ("2.7", []),
+                ("3.6", []),
+            ]),
+            ("9.2", [("3.6", [])]),
+            ("10", [("3.6", [])]),
+        ]),
+    ]),
+]
+
+
+def get_root():
+    return TopLevelNode("Pytorch Builds", CONFIG_TREE_DATA)
+
+
+def gen_tree():
+    root = get_root()
+    configs_list = conf_tree.dfs(root)
+    return configs_list
+
+
+class TopLevelNode(ConfigNode):
+    def __init__(self, node_name, config_tree_data):
+        super(TopLevelNode, self).__init__(None, node_name)
+
+        self.config_tree_data = config_tree_data
+
+    def get_children(self):
+        return [DistroConfigNode(self, d, p) for (d, p) in self.config_tree_data]
+
+
+class DistroConfigNode(ConfigNode):
+    def __init__(self, parent, distro_name, py_tree):
+        super(DistroConfigNode, self).__init__(parent, distro_name)
+
+        self.py_tree = py_tree
+        self.props["distro_name"] = distro_name
+
+    def get_children(self):
+        return [PyVerConfigNode(self, k, v) for k, v in self.py_tree]
+
+
+class PyVerConfigNode(ConfigNode):
+    def __init__(self, parent, pyver, compiler_tree):
+        super(PyVerConfigNode, self).__init__(parent, pyver)
+
+        self.compiler_tree = compiler_tree
+        self.props["pyver"] = pyver
+
+    def get_children(self):
+
+        if self.find_prop("distro_name") == "trusty":
+            return [CompilerConfigNode(self, v, xla_options) for (v, xla_options) in self.compiler_tree]
+        else:
+            return []
+
+
+class CompilerConfigNode(ConfigNode):
+    def __init__(self, parent, compiler_name, xla_options):
+        super(CompilerConfigNode, self).__init__(parent, compiler_name)
+
+        self.xla_options = xla_options
+
+    def get_children(self):
+        return [XlaConfigNode(self, v) for v in self.xla_options]
+
+
+class XlaConfigNode(ConfigNode):
+    def __init__(self, parent, xla_enabled):
+        super(XlaConfigNode, self).__init__(parent, "XLA=" + str(xla_enabled))
+
+        self.xla_enabled = xla_enabled
+
+    def get_children(self):
+        return []
+
+
+BUILD_ENV_LIST = [
+    Conf("trusty", ["py2.7.9"]),
+    Conf("trusty", ["py2.7"]),
+    Conf("trusty", ["py3.5"]),
+    Conf("trusty", ["py3.6", "gcc4.8"]),
+    Conf("trusty", ["py3.6", "gcc5.4"]),
+    Conf("trusty", ["py3.6", "gcc5.4"], is_xla=True),
```

Comment:
This is_xla is nice. We should expand this to cover all the USE_* that we want to test for tesnrorrt, leveldb, lmdb, redis, opencv, mkldnn, ideep, etc.